src/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -11,29 +11,34 @@ namespace System.Net.Http
 {
     public class StreamContent : HttpContent
     {
-        private const int DefaultBufferSize = 4096;
-
         private Stream _content;
         private int _bufferSize;
         private bool _contentConsumed;
         private long _start;
 
         public StreamContent(Stream content)
-            : this(content, DefaultBufferSize)
         {
+            // Indicate that we should use default buffer size by setting size to 0.
+            InitializeContent(content, 0);
         }
 
         public StreamContent(Stream content, int bufferSize)
         {
-            if (content == null)
-            {
-                throw new ArgumentNullException(nameof(content));
-            }
             if (bufferSize <= 0)
             {
                 throw new ArgumentOutOfRangeException(nameof(bufferSize));
             }
 
+            InitializeContent(content, bufferSize);
```

Comment:
Nit: not sure if we care, but this will change what exception you receive if you do `new StreamContent(null, -1)`... previously you'd receive an ArgumentNullException, now you'll get an ArgumentOutOfRangeException.

Reply:
I would prefer to keep the behavior the same....so we should check explicitly for 'null' before calling the InitializeContent method.

Reply:
I did consider that, but wasn't sure what stance we took on that kind of change. I'd be happy to just move the error checks out of the new `InitializeContent` method if we want to maintain strict compatibility.

Reply:
I think we should always lean on the side of preserving behavior especially in terms of what exceptions we throw.